src/Phinx/Console/Command/Init.php

```diff
@@ -53,55 +57,79 @@ protected function configure()
     /**
      * Initializes the application.
      *
-     * @param \Symfony\Component\Console\Input\InputInterface $input
-     * @param \Symfony\Component\Console\Output\OutputInterface $output
-     * @throws \RuntimeException
-     * @throws \InvalidArgumentException
+     * @param InputInterface  $input  Interface implemented by all input classes.
+     * @param OutputInterface $output Interface implemented by all output classes.
+     *
+     * @throws RuntimeException
+     * @throws InvalidArgumentException
      * @return void
      */
     protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $path = $this->resolvePath($input);
+        $this->writeConfig($path);
+
+        $output->writeln("<info>created</info> {$path}");
+    }
+
+    /**
+     * Return valid $path for Phing's config file.
+     *
+     * @param InputInterface $input Interface implemented by all input classes.
+     *
+     * @return string
+     */
+    protected function resolvePath(InputInterface $input)
     {
         // get the migration path from the config
         $path = $input->getArgument('path');
 
-        if ($path === null) {
-            $path = getcwd();
+        // Fallback
+        if (empty($path)) {
+            $path = getcwd() . DIRECTORY_SEPARATOR . self::FILE_NAME;
         }
 
-        $path = realpath($path);
-
-        if (!is_writable($path)) {
-            throw new \InvalidArgumentException(sprintf(
-                'The directory "%s" is not writable',
-                $path
-            ));
```

Comment:
I removed this check because I thought that it was used as a replacement of `is_dir` as I currently do. Moreover, unlike before, now we have to check the non existence of a file (e.g. `phing.yml`).

Reply:
But wouldn't `is_dir()` return `true` even if you don't have write permissions for the directory? I thought we could keep the message, as `$dirname` would be a directory always.

Reply:
I restored the `is_writable` check, it's located in `writeConfig()` method.